crates/autopilot/src/solvable_orders.rs

```diff
@@ -238,18 +238,20 @@ impl SolvableOrdersCache {
         let removed = counter.record(&orders);
         filtered_order_events.extend(removed);
 
-        // spawning a background task since `order_events` table insert operation takes
-        // a while and the result is ignored.
-        self.persistence.store_order_events(
-            invalid_order_uids.iter().map(|id| domain::OrderUid(id.0)),
-            OrderEventLabel::Invalid,
-        );
-        self.persistence.store_order_events(
-            filtered_order_events
-                .iter()
-                .map(|id| domain::OrderUid(id.0)),
-            OrderEventLabel::Filtered,
-        );
+        if store_events {
```

Comment:
We want only 1 instance to write these to the DB and unfortunately it's in what's otherwise a read-only path, so I put it behind a feature flag so that only the leader writes here.